sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyAsyncClient.java

```diff
@@ -167,44 +183,44 @@
      *
      * {@codesnippet com.azure.security.keyvault.keys.async.keyclient.createKey#keyCreateOptions}
      *
-     * @param keyCreateOptions The key configuration object containing information about the key being created.
-     * @return A {@link Mono} containing the {@link Key created key}.
+     * @param createKeyOptions The key configuration object containing information about the key being created.
+     * @return A {@link Mono} containing the {@link KeyVaultKey created key}.
      * @throws NullPointerException if {@code keyCreateOptions} is {@code null}.
      * @throws ResourceModifiedException if {@code keyCreateOptions} is malformed.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Key> createKey(KeyCreateOptions keyCreateOptions) {
+    public Mono<KeyVaultKey> createKey(CreateKeyOptions createKeyOptions) {
         try {
-            return createKeyWithResponse(keyCreateOptions).flatMap(FluxUtil::toMono);
+            return createKeyWithResponse(createKeyOptions).flatMap(FluxUtil::toMono);
         } catch (RuntimeException ex) {
             return monoError(logger, ex);
         }
     }
 
-    Mono<Response<Key>> createKeyWithResponse(KeyCreateOptions keyCreateOptions, Context context) {
-        Objects.requireNonNull(keyCreateOptions, "The key create options parameter cannot be null.");
+    Mono<Response<KeyVaultKey>> createKeyWithResponse(CreateKeyOptions createKeyOptions, Context context) {
+        Objects.requireNonNull(createKeyOptions, "The key create options parameter cannot be null.");
         KeyRequestParameters parameters = new KeyRequestParameters()
-            .setKty(keyCreateOptions.getKeyType())
-            .setKeyOps(keyCreateOptions.keyOperations())
-            .setKeyAttributes(new KeyRequestAttributes(keyCreateOptions));
-        return service.createKey(endpoint, keyCreateOptions.getName(), API_VERSION, ACCEPT_LANGUAGE, parameters,
+            .setKty(createKeyOptions.getKeyType())
```

Comment:
```
 .setKeyType(createKeyOptions.getKeyType())
```

Reply:
internal class, mapping to swagger/REST API
Best to keep it same as swagger/REST API